src/yz_kv.erl

```diff
@@ -396,6 +405,19 @@ cleanup(_, _) ->
 
 %% @private
 %%
+%% @doc Cleanup tombstones and siblings accordingly... cleanup/3
+%% TODO: make this work for allow_mult=true case
+-spec cleanup([doc()], bkey(), values()) -> [{bkey, bkey()}].
+cleanup([], _BKey, _Values) ->
+    [];
+cleanup([{doc, Fields}|T], BKey, [VH|VT]) ->
```

Comment:
@jonmeredith good point.. actually was already doing this... but will do when i bring this over to the main PR.